src/PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.cpp

```diff
@@ -0,0 +1,192 @@
+        0.25 / (shear_modulus * M_PI) *
+        integration(
+            [&r, &z, &surface_term_z, this](const double k) noexcept {
+              return gsl_sf_bessel_J0(k * r) *
+                     exp(-k * z - square(k * beam_width_ / 2.)) *
+                     (surface_term_z + k * z);
+            },
+            lower_boundary, absolute_tolerance_, relative_tolerance_);
+    get<2>(result)[i] = result_z;
+  }
+  return {std::move(result)};
+}
+
+tuples::TaggedTuple<Tags::Strain<3>> HalfSpaceMirror::variables(
+    const tnsr::I<DataVector, 3>& x, tmpl::list<Tags::Strain<3>> /*meta*/) const
+    noexcept {
+  const double shear_modulus = constitutive_relation_.shear_modulus();
+  const double lame_parameter = constitutive_relation_.lame_parameter();
+  auto strain = make_with_value<tnsr::ii<DataVector, 3>>(x, 0.);
+  const auto radius = sqrt(square(get<0>(x)) + square(get<1>(x)));
+  const integration::GslQuadAdaptive<
+      integration::GslIntegralType::UpperBoundaryInfinite>
+      integration{integration_intervals_};
+  const double lower_boundary = 0.;
+  const size_t num_points = get<0>(x).size();
+  for (size_t i = 0; i < num_points; i++) {
+    const double r = radius[i];
+    const double z = get<2>(x)[i];
+
+    const double surface_term_trace =
+        -2. * shear_modulus / (lame_parameter + shear_modulus);
+    const double trace_term =
+        0.25 / (shear_modulus * M_PI) *
+        integration(
+            [&r, &z, &surface_term_trace, this](const double k) noexcept {
+              return k * gsl_sf_bessel_J0(k * r) *
+                     exp(-k * z - square(k * beam_width_ / 2.)) *
+                     surface_term_trace;
+            },
+            lower_boundary, absolute_tolerance_, relative_tolerance_);
+
+    const double surface_term_zz =
+        shear_modulus / (lame_parameter + shear_modulus);
+    const double strain_zz =
+        -0.25 / (shear_modulus * M_PI) *
+        integration(
+            [&r, &z, &surface_term_zz, this](const double k) noexcept {
+              return k * gsl_sf_bessel_J0(k * r) *
+                     exp(-k * z - square(k * beam_width_ / 2.)) *
+                     (surface_term_zz + k * z);
+            },
+            lower_boundary, absolute_tolerance_, relative_tolerance_);
+
+    if (r >= 100 * std::numeric_limits<double>::epsilon()) {
+      const double strain_rz =
+          -0.25 / (shear_modulus * M_PI) *
+          integration(
+              [&r, &z, this](const double k) noexcept {
+                return k * gsl_sf_bessel_J1(k * r) *
+                       exp(-k * z - square(k * beam_width_ / 2.)) * (k * z);
+              },
+              lower_boundary, absolute_tolerance_, relative_tolerance_);
+
+      const double surface_term_r = 1. - (lame_parameter + 2. * shear_modulus) /
+                                             (lame_parameter + shear_modulus);
+      const double displacement_r =
+          0.25 / (shear_modulus * M_PI) *
```

Comment:
This is duplicated code... can this be made an anonymous namespace function? Also, looks like you don't have the 1/r term here like you did when you previously calculated displacement_r

Reply:
I have moved the integrand to a separate functions that is called both in the evaluation of Displacement and Strain.